aries/agents/rust/mediator/src/aries_agent/mod.rs

```diff
@@ -151,11 +151,17 @@ impl<T: BaseWallet + 'static, P: MediatorPersistence> Agent<T, P> {
             .as_deref()
             .ok_or("Anonymous sender can't be authenticated")?
             .to_owned();
-        let (_sr_no, account_name, our_signing_key, did_doc_json) =
-            self.persistence.get_account_details(&auth_pubkey).await?;
-        let diddoc =
-            serde_json::from_value::<AriesDidDoc>(did_doc_json).map_err(string_from_std_error)?;
-        Ok((account_name, auth_pubkey, our_signing_key, diddoc))
+        let account_details = self
+            .persistence
+            .get_account_details(&auth_pubkey)
+            .await
+            .map_err(string_from_std_error)?;
+        Ok((
+            account_details.account_name,
+            account_details.auth_pubkey,
+            account_details.our_signing_key,
+            account_details.their_did_doc,
+        ))
```

Comment:
out of scope comment: should return the account details type rather than tuple

Reply:
Done